Libs/Python/ShapeworksPython.cpp

```diff
@@ -1116,10 +1116,60 @@ PYBIND11_MODULE(shapeworks_py, m)
        "point"_a)
 
   .def("geodesicDistance",
-       &Mesh::geodesicDistance,
+       py::overload_cast<int, int>(&Mesh::geodesicDistance),
        "computes geodesic distance between two vertices (specified by their indices) on mesh",
        "source"_a, "target"_a)
 
+  .def("geodesicDistance",
+       [](Mesh &mesh, const std::vector<double> p) -> decltype(auto) {
+          auto array = mesh.geodesicDistance(Point({p[0], p[1], p[2]}));
+          const auto shape = std::vector<size_t>{static_cast<unsigned long>(array->GetNumberOfTuples()),
+                                                 static_cast<unsigned long>(array->GetNumberOfComponents()),
+                                                 1};
+          auto vtkarr = vtkSmartPointer<vtkDoubleArray>(vtkDoubleArray::New());
+          vtkarr->SetNumberOfValues(array->GetNumberOfValues());
+
+          // LOTS of copying going on here, see github #903
+          array->GetData(0, array->GetNumberOfTuples()-1,
+                         0, array->GetNumberOfComponents()-1,
+                         vtkarr);                               // copy1
+          return py::array(py::dtype::of<double>(),
+                         shape,
+                         vtkarr->GetVoidPointer(0));          // copy2
+       },
+       "computes geodesic distance between a point (landmark) and each vertex on mesh",
+       "landmark"_a)
+
+  .def("geodesicDistance",
+       [](Mesh &mesh, const std::vector<std::vector<double>> p) -> decltype(auto) {
+          std::vector<Point> points;
+          for (int i=0; i<p.size(); i++)
+          {
+            points.push_back(Point({p[i][0], p[i][0], p[i][2]}));
+          }
+          auto array = mesh.geodesicDistance(points);
+          const auto shape = std::vector<size_t>{static_cast<unsigned long>(array->GetNumberOfTuples()),
+                                                 static_cast<unsigned long>(array->GetNumberOfComponents()),
+                                                 1};
+          auto vtkarr = vtkSmartPointer<vtkDoubleArray>(vtkDoubleArray::New());
+          vtkarr->SetNumberOfValues(array->GetNumberOfValues());
+
+          // LOTS of copying going on here, see github #903
+          array->GetData(0, array->GetNumberOfTuples()-1,
+                         0, array->GetNumberOfComponents()-1,
+                         vtkarr);                               // copy1
+          return py::array(py::dtype::of<double>(),
+                         shape,
+                         vtkarr->GetVoidPointer(0));          // copy2
+       },
+       "computes geodesic distance between a set of points (curve) and all vertices on mesh",
+       "curve"_a)
+
+  .def("distance",
+       &Mesh::distance,
+       "computes surface to surface distance",
```

Comment:
I suggest we tell the user in the help text the methods of computing distance that can be used.
I also recommend changing their Python enums to use CamelCase just as all the other enums and functions here. 
i.e., PointToPoint and CellToCell rather than POINT_TO_POINT and CELL_TO_CELL (around line 950).

Reply:
done